python/ray/dag/dag_node_operation.py

```diff
@@ -0,0 +1,102 @@
+from functools import total_ordering
+from enum import Enum
+from ray.util.annotations import DeveloperAPI
+import ray
+
+
+@DeveloperAPI
```

Comment:
I don't think we need to make this DeveloperAPI?

Reply:
IIRC, the lint check in the CI will fail if we don't add DeveloperAPI. Let me try it.

Reply:
Updated https://github.com/ray-project/ray/pull/46911/commits/383b877cb353c958e12e08ea0af2c4d0f23b0519.

Reply:
<img width="680" alt="image" src="https://github.com/user-attachments/assets/c8bbf04d-3d3b-4d69-90d4-84c515b530a3">

Reply:
The CI failed https://buildkite.com/ray-project/microcheck/builds/3914#01912ff2-072f-4f91-8c4b-e071cee5dd60

Reply:
I will add it back

Reply:
If you prepend an underscore to the name to indicate that it is a private class, you should not get the warning anymore.

Reply:
> If you prepend an underscore to the name to indicate that it is a private class

Good to know. Thanks! Btw, what's the definition of `DeveloperAPI`? The annotation's doc string says:

"Developer APIs are lower-level methods explicitly exposed to advanced Ray users and library developers. Their interfaces may change across minor Ray releases."

I found that we also add annotations to functions like `do_allocate_channel`. I think it isn't exposed to Ray users or Ray library developers. Am I misunderstanding the definition?

https://github.com/ray-project/ray/blob/9e9893fd381a5adbb77f0d559814ddfc36eb22c2/python/ray/dag/compiled_dag_node.py#L53-L54

Reply:
I think this was from back when I did not know you could do the underscore thing :D Feel free to update it.

Reply:
Open an issue: https://github.com/ray-project/ray/issues/47048

Reply:
Updated:
* https://github.com/ray-project/ray/pull/46911/commits/fadf05f14c17d756a156785765701f28939fc539
* https://github.com/ray-project/ray/pull/46911/commits/9895d8638cd53f53e6802b48ab0ea66c54ed12cb